### PR TITLE
(1) Add Data Sources docs

### PR DESCRIPTION
**Pull request by sylviamoss**
This is the docs for https://github.com/hashicorp/packer/pull/10440. 

It will have conflicts with https://github.com/hashicorp/packer/pull/10486 but it should not be hard to fix them. [done ✔️ ]

### CODE REVIEW

**Comment by SwampDragons** [comment]
I fixed the conflicts for you since I merged those other website changes. The only thing you have left to do is add the 

https://packer.io/docs/datasources 

page. You probably want an index landing page, and to add the amazon-ami datasource to the side bar.

**Comment by sylviamoss** [comment]
Nice! Thanks for rebasing! 🤗 

> The only thing you have left to do is add the
> https://packer.io/docs/datasources

This is being done in the [amazon-ami PR](https://github.com/hashicorp/packer/pull/10467/files#diff-4da40f726300b7d33766ea8e46cd5b9b00a15e4e89a9837a71ed88bbf2b0a340R1). I wrote the amazon-ami docs first so that's why it ended up there.

File: website/content/docs/templates/hcl_templates/datasources.mdx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+The data block creates a data instance of the given _type_ (first block label) and _name_ (second block label).
+The combination of the type and name must be unique within a configuration.
+
+Within the block (the `{ }`) is the configuration for the data instance. The configuration is dependent on the type,
+and is documented for each data source in the [data sources](/docs/datasources) section.
+
+A data source can output one or more attributes, which can be used by adding their key name to the data source unique
+identifier, like `data.<TYPE>.<NAME>.<ATTRIBUTE>`.
+
+The output from the `amazon-ami.example` above can be access as follows:
```
Comment (azr) [inline website/content/docs/templates/hcl_templates/datasources.mdx:46]:
```suggestion
The output from the `amazon-ami.example` above can be accessed as follows:
```

**Review by azr — APPROVED** [review]
Nice one LGTM, just had some minor tweaks

**Review by azr — APPROVED** [review]
👍🏼

**Review by SwampDragons — APPROVED** [review]

**Pull request merged**